types/valuerenderer/valuerenderer.go

```diff
@@ -0,0 +1,263 @@
+package valuerenderer
+
+import (
+	"context"
+	"errors"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"golang.org/x/text/language"
+	"golang.org/x/text/message"
+)
+
+type ValueRenderer interface {
+	Format(interface{}) (string, error)
+	Parse(string) (interface{}, error)
+}
+
+// create default value rreenderer in CLI and then get context from CLI
+type DefaultValueRenderer struct {
+	// /string is denom that user sents
+	bankKeeper keeper.BaseKeeper // define in test only //convert DenomUnits to Display units
+	metaData   banktypes.Metadata
+}
+
+func NewDefaultValueRenderer(bk keeper.BaseKeeper) DefaultValueRenderer {
```

Comment:
@clevinson I set bankKeeper on `DefaultValueRenderer` to call `DenomMetadata(ctx, req)` on L47